sklearn/preprocessing/_encoders.py

```diff
@@ -848,3 +852,246 @@ def inverse_transform(self, X):
             X_tr[:, i] = self.categories_[i][labels]
 
         return X_tr
+
+
+class UnaryEncoder(BaseEstimator, TransformerMixin):
+    """Encode ordinal integer features using a unary scheme.
+
+    This encoder transforms each ordinal feature with ``m`` possible values
+    into ``m - 1`` binary features, where the ith feature is active if ``x >
+    i``. The input to this transformer should be a matrix of non-negative
+    integers, denoting the values taken on by the ordinal features.
+
+    This encoding may be needed for feeding ordinal features to many
+    scikit-learn estimators, notably linear models and kernel-based models
+    like SVMs with the standard kernels.
+    This transformation is unlikely to help when using with tree-based models,
+    since those already work on the basis of a particular feature value being
+    less or greater than a threshold, unlike linear and kernel-based models.
+
+    This encoder encodes all of the features. To only encode a subset of the
+    features, use :class:`ColumnTransformer
+    <sklearn.compose.ColumnTransformer>`.
+
+    Read more in the :ref:`User Guide <unary_encoding>`.
+
+    Parameters
+    ----------
+    n_values : 'auto', int or array of ints, optional (default='auto')
+        Number of values per feature.
+
+        - 'auto' : determine value range from training data.
+        - int : number of ordinal values per feature.
+                Each feature value should be in ``range(n_values)``
+        - array : ``n_values[i]`` is the number of ordinal values in
+                  ``X[:, i]``. Each feature value should be
+                  in ``range(n_values[i])``
+
+    dtype : number type, optional (default=np.float)
+        Desired dtype of output.
+
+    sparse : boolean, optional (default=False)
```

Comment:
I'm wondering whether we should change the default to True

Reply:
I would say no, since if the categories are evenly distributed the transformed matrix should have more ones than zeros.

Reply:
> I would say no, since if the categories are evenly distributed the transformed matrix should have more ones than zeros.

Agree, let's keep the default.